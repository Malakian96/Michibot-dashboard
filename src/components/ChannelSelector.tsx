import { Channel } from "./Main";

interface ChannelSelectorProps {
  channels: Channel[];
  selectedChannel: string;
  setSelectedChannel: (channelId: string) => void;
}

const ChannelSelector = ({ channels, selectedChannel, setSelectedChannel }: ChannelSelectorProps) => {
    return (
      <select
        value={selectedChannel}
        onChange={(e) => setSelectedChannel(e.target.value)}
        disabled={!channels.length}
        className="bg-discord-dark-light text-white px-4 py-2 rounded-md mb-4"
      >
        <option value="">Select a voice channel</option>
        {channels.map((channel: Channel) => (
          <option key={channel.id} value={channel.id}>
            {channel.name}
          </option>
        ))}
      </select>
    );
  };
  
  export default ChannelSelector;
  