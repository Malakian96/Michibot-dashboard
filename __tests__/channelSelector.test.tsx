import { render, screen, fireEvent } from '@testing-library/react';
import ChannelSelector from '../src/components/ChannelSelector';

const mockChannels = [
  { id: '1', name: 'Channel 1' },
  { id: '2', name: 'Channel 2' },
];

const mockSetSelectedChannel = jest.fn();

describe('ChannelSelector', () => {
  it('renders correctly', () => {
    render(
      <ChannelSelector
        channels={mockChannels}
        selectedChannel=""
        setSelectedChannel={mockSetSelectedChannel}
      />,
    );
    expect(screen.getByText('Select a voice channel')).toBeInTheDocument();
  });

  it('displays the correct options', () => {
    render(
      <ChannelSelector
        channels={mockChannels}
        selectedChannel=""
        setSelectedChannel={mockSetSelectedChannel}
      />,
    );
    expect(screen.getByText('Channel 1')).toBeInTheDocument();
    expect(screen.getByText('Channel 2')).toBeInTheDocument();
  });

  it('calls setSelectedChannel on change', () => {
    render(
      <ChannelSelector
        channels={mockChannels}
        selectedChannel=""
        setSelectedChannel={mockSetSelectedChannel}
      />,
    );
    fireEvent.change(screen.getByRole('combobox'), { target: { value: '1' } });
    expect(mockSetSelectedChannel).toHaveBeenCalledWith('1');
  });
});

describe('ChannelSelectorSnapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ChannelSelector
        channels={mockChannels}
        selectedChannel=""
        setSelectedChannel={mockSetSelectedChannel}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
