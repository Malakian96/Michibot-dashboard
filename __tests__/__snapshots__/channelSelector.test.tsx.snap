// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelSelectorSnapshot matches snapshot 1`] = `
<div>
  <select
    class="bg-discord-dark-light text-white px-4 py-2 rounded-md mb-4"
  >
    <option
      value=""
    >
      Select a voice channel
    </option>
    <option
      value="1"
    >
      Channel 1
    </option>
    <option
      value="2"
    >
      Channel 2
    </option>
  </select>
</div>
`;
